Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift

```diff
@@ -75,200 +91,1089 @@ public class ComponentViewController: UIView {
-                displayText.labelText = blockFieldString ?? "Demo Text"
-                displayText.textColor = UIColor(hexString: blockTextColor)
-                displayText.isTextBold = blockTextWeight == "bold" ? true : false
-                displayText.isTextItalic = blockTextStyle == "italic" ? true : false
-                displayText.textAlignment = blockTextAlignment.textAlignmentFromStringValue()
-                displayText.frame = CGRect(x: 20, y: 10, width: tableView.bounds.width - 40, height: tableView.rowHeight)
-                cellView.append(displayText)
-                
-            } else if componentType[i] == "number" {
-                
-                // MARK: Number Function Call From Package
-                let number = NumberField()
-                tableView.rowHeight = 100
-                number.titleText = componentHeaderText[i]
-                number.currentPage = numberFieldString ?? 0
-                number.numberField.text = "\(numberFieldString ?? 0)"
-                number.toolTipTitle = "Testing Number"
-                number.toolTipDescription = "Should be same person that signs signature field. Don’t forget to capture."
-                number.frame = CGRect(x: 10, y: 10, width: tableView.bounds.width - 20, height: tableView.rowHeight)
-                cellView.append(number)
-                
-            } else if componentType[i] == "chart" {
-                
-                // MARK: Chart Function Call From Package
-                let chart = Chart()
-                tableView.rowHeight = 380
-                chart.performanceGraphLabel.labelText = componentHeaderText[i]
-                chart.toolTipTitle = "Testing Chart"
-                chart.toolTipDescription = "Should be same person that signs signature field. Don’t forget to capture."
-                chart.frame = CGRect(x: 10, y: 0, width: tableView.bounds.width - 20, height: tableView.rowHeight)
-                cellView.append(chart)
-                
-            } else if componentType[i] == "table" {
-                
-                // MARK: Chart Function Call From Package
-                let table = Table()
-                table.toolTipTitle = "Testing Table"
-                table.titleLabel.labelText = componentHeaderText[i]
-                table.toolTipDescription = "Should be same person that signs signature field. Don’t forget to capture."
-                let height = (componentHeaderText[indexPath.row].height(withConstrainedWidth: frame.width, font: table.titleLabel.font))
-                tableView.rowHeight = 220 + height + 20
-                tableHeight = tableView.rowHeight
-                table.frame = CGRect(x: 10, y: 0, width: tableView.bounds.width - 20, height: tableView.rowHeight)
-                cellView.append(table)
-            }
+        let value = joyDocFieldData[indexPath.row].value
+        
+        if joyDocFieldData[indexPath.row].type == "text" {
```

Comment:
2 things here:
1. We should use a switch statement for things like this
2. These field types `text`, `date` ect. Can we make these a global constants file that can be used throughout like our `constants/FieldTypes.js`. So these would be `== FieldTypes.date` ect...

Reply:
Done.